### ISSUE TITLE
Incorrect language redirection in addons.mozilla.org

### ISSUE THREAD

**Issue by yookoala** [mozilla/addons-server#6239]
### Describe the problem and steps to reproduce it:

Website should follow browser's language priority when doing redirection / content detection. The "Accepted-Language" header include priority for the language in it. All redirection or locale detection should respect the priority specified there.

For example, when provided with this "`Accepted-Language`" header:
```
zh-HK,zh-TW;q=0.8,en-US;q=0.5,en;q=0.3"
```
a website should:

1. find "`zh-HK`" (Traditional Chinese, Hong Kong) and "`zh-TW`" (Traditional Chinese, Taiwan) first. If exists, redirect / show the content in that language.
2. If not exists, find "`en-US`". redirect / show the content in that language.
3. If not exists, find "`en`". redirect / show the content in that language.

### What happened?

If you supply a request to [addons.mozilla.org/firefox](https://addons.mozilla.org/firefox) with the said header:
```
zh-HK,zh-TW;q=0.8,en-US;q=0.5,en;q=0.3"
```

it will redirect you to [addons.mozilla.org/zh-CN/firefox](https://addons.mozilla.org/zh-CN/firefox). 

### What did you expect to happen?

That is not an expected response. It is wrong in multiple levels:

1. Even there is no "`zh-HK`" in the site, there is "`zh-TW`". From language priority, it should be shown.
2. "`zh-CN`" is never a part of the language preference. If should not be shown, or it should be shown as a fallback only if both "`zh-HK`" and "`zh-TW`" are absent.
3. Event without language priority to "`zh-TW`", the fallback to "`zh-HK`" should always be "`zh-TW`" (both Traditional Chinese).

**Comment by eviljeff** [comment]
What's happening is we have fallback code that takes an unsupported long code (`zh-HK`) and converts it into a short code (`zh`) which then converts to the default for that short code (`zh-CN`).  Then, because a locale has been 'found', it ignores the subsequent locale codes in the accept header.  (It can also be replicated with `pt-XX,pt-BR` which redirects to `pt-PT` instead)

For zh-* see #5829 also

**Comment by yookoala** [comment]
Is there anyway to override this redirection? The fallback of `zh-HK` or `zh-Hant-HK` should have been `zh-Hant` (if made available in #5811).

**Comment by eviljeff** [comment]
In the end `zh-Hant` wasn't made available in #5811, as #5829 mentions.

There isn't a way to override the redirection, we need to fix the logic.  In the meantime you can visit https://addons.mozilla.org/zh-TW/firefox directly.

**Comment by yookoala** [comment]
The issue is still not fixed. Please re-open this issue.
